api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -63,6 +72,80 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
 		for (Order savedOrder : savedOrderGroup.getOrders()) {
 			assertNull("Order is not saved as a part of Order Group", savedOrder.getOrderId());
 		}
-		
+	}
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByEncounter_shouldFailGivenNullEncounter() {
+		dao.getOrderGroupsByEncounter(null);
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByPatient_shouldFailGivenNullPatient() {
+		dao.getOrderGroupsByPatient(null);
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFromAnEncounter() {
+		Encounter existingEncounter = Context.getEncounterService().getEncounter(4);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter);
+
+		assertEquals(1, ordergroups1.size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFrommMultipleEncounters() {
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		Encounter existingEncounter2 = Context.getEncounterService().getEncounter(6);
```

Comment:
What is the use of this test?

Reply:
the purpose of this test is to create instances of multiple encounters and they do exist in the existing code